### PR TITLE
View for bus trips implemented

### LINKED ISSUE #17: Bus trips basic functionality + live notification
Will also implement the basic model of the trips and try and make the app awake to send a notification at the given time

### PR DESCRIPTION
**Pull request by JoshuaJLi**
Ironed out process for handling the different schedules and finding when the next upcoming trip is. Also displays on the fragment

image

### CODE REVIEW

File: app/src/main/java/ca/wheresthebus/MainDBViewModel.kt [hunk 1]
```diff
@@ -165,4 +166,9 @@ class MainDBViewModel : ViewModel() {
         }
         return modelFactory.toRoute(route[0]) ?: null
     }
+
+    fun getTrips(): ArrayList<ScheduledTrip> {
+        val trips = ArrayList<ScheduledTrip>()
```
Comment (bryanjhdang) [inline app/src/main/java/ca/wheresthebus/MainDBViewModel.kt:171]: change this to actual scheduledtrip db data

File: app/src/main/java/ca/wheresthebus/ui/trips/TripsFragment.kt [hunk 2]
```diff
@@ -22,21 +37,82 @@ class TripsFragment : Fragment() {
...
+    }
+
     override fun onDestroyView() {
         super.onDestroyView()
         _binding = null
     }
+
+    companion object {
+        object TripType {
+            const val ACTIVE = "active"
```
Comment (bryanjhdang) [inline app/src/main/java/ca/wheresthebus/ui/trips/TripsFragment.kt:113]: there seems to be a lot of ACTIVE / INACTIVE constants being made throughout the program, could make have a Globals file?

**Review by bryandangdev — APPROVED** [review]
The code is mostly good, just some concerns that there seems be be a lot of instances of ACTIVE / INACTIVE / etc. variables being made for each of the classes that require them. Maybe take a look into the code to check we can reduce code duplication, if any.

**Comment by bryanjhdang** [comment]
LGTM

**Pull request merged**